### PR TITLE
Mention env var when using API Token and `whoami`/`log(out|in)` commands

### PR DESCRIPTION
**Pull request by dom96**
I ran into this while trying to log in to my other account. Didn't realise wrangler uses this env var, so was confused why I couldn't logout.

**What this PR solves / how to test:**

```
$ pnpm run build --filter wrangler
$ pnpm run --filter wrangler start whoami
$ pnpm run --filter wrangler start logout
$ pnpm run --filter wrangler start login
```

Screenshot 2024-02-05 at 18 47 08
Screenshot 2024-02-05 at 18 47 47
Screenshot 2024-02-05 at 18 56 26

**Author has addressed the following:**

- Tests
  - [ ] Included
  - [x] Not necessary because: simple change
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] Included
  - [ ] Not necessary because:
- Associated docs
  - [ ] Issue(s)/PR(s):
  - [x] Not necessary because: simple change

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Hey @dom96 - thanks for improving this DX.

- The code is failing a formatting check. Can you try running `pnpm fix` in the root of the repo to fix this?
- The code is failing unit tests - a snapshot [need updating](https://github.com/cloudflare/workers-sdk/actions/runs/7789470965/job/21241164299?pr=4925#step:8:1453). Can you try running `pnpm -F wrangler test:ci -u`

**Comment by petebacondarwin** [comment]
Let's see how the CI pans out. I can look into any random failures.

File: packages/wrangler/src/__tests__/whoami.test.tsx [hunk 1]
```diff
@@ -161,14 +161,6 @@ describe("getUserInfo()", () => {
...
-		expect(std.warn).toMatchInlineSnapshot(`
-		      "[33m▲ [43;33m[[43;30mWARNING[43;33m][0m [1mIt looks like you have used Wrangler v1's \`config\` command to login with an API token.[0m
-
-		        This is no longer supported in the current version of Wrangler.
-		        If you wish to authenticate via an API token then please set the \`CLOUDFLARE_API_TOKEN\`
-		        environment variable.
-
-		      "
-	    `);
+		expect(std.warn).toMatchInlineSnapshot(`""`);
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/whoami.test.tsx:164]: This needs fixing...

**Review by petebacondarwin — APPROVED** [review]

**Comment by dom96** [comment]
Thank you for the help in getting this merged in!

**Pull request merged**